pygen/pygen_src/riscv_data_page_gen.py

```diff
@@ -0,0 +1,87 @@
+"""
+Copyright 2020 Google LLC
+Copyright 2020 PerfectVIPs Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+"""
+
+import logging
+import vsc
+import random
+from collections import defaultdict
+from pygen_src.riscv_instr_pkg import pkg_ins, data_pattern_t
+from pygen_src.riscv_instr_gen_config import cfg
+
+
+@vsc.randobj
+class riscv_data_page_gen:
+    def __init__(self):
+        self.data_page_str = []
+        self.mem_region_setting = defaultdict(list)
+
+    def gen_data(self, idx, pattern, num_of_bytes, data):
+        temp_data = 0
+        data = [0] * num_of_bytes
+        for i in range(len(data)):
+            if pattern == data_pattern_t.RAND_DATA:
+                temp_data = random.randrange(0, (2**8) - 1)
+                data[i] = temp_data
+            elif pattern == data_pattern_t.INCR_VAL:
+                data[i] = (idx + i) % 256
+        return data
+
+    def gen_data_page(self, hart_id, pattern, is_kernel=0, amo=0):
+        tmp_str = ""
+        temp_data = []
+        tmp_data = []
+        page_cnt = 0
+        page_size = 0
+        self.data_page_str.clear()
+        if is_kernel:
+            self.mem_region_setting = cfg.s_mem_region
+        elif amo:
+            self.mem_region_setting = cfg.amo_region
+        else:
+            self.mem_region_setting = cfg.mem_region
+        for i in range(len(self.mem_region_setting)):
+            logging.info("Generate data section: %0s size:0x%0x xwr:0x%0x",
```

Comment:
nit: Can we use `format` instead of `%`? It's better to be consistent throughout different stages (we're using `format` a few lines later as well).